src/client.rs

```diff
@@ -501,12 +505,37 @@ fn main() {
     .filter(Some("tokio_reactor"), LevelFilter::Off)
     .init();
 
-    let addr = env::args().nth(1).unwrap_or("127.0.0.1:12345".to_owned());
-    let addr = addr.parse::<SocketAddr>()
-       .unwrap_or_else(|e| {
-                    error!("failed to parse address {:?}: {:?}", addr, e);
-                    exit(1);
-                });
+    let has_port_re = Regex::new(r":\d{1,5}$").unwrap(); // match a colon followed by number up to 5 digits (16-bit port)
+    let mut server_str = env::args().nth(1).unwrap_or("localhost".to_owned());
+    // if no port, add the default port
+    if !has_port_re.is_match(&server_str) {
+        debug!("Appending default port to {:?}", server_str);
+        server_str = format!("{}:{}", server_str, DEFAULT_PORT);
+    }
+
+    // synchronously resolve DNS because... why not?
+    trace!("Resolving {:?}...", server_str);
+    let addr_vec = tokio_dns::resolve_sock_addr(&server_str[..]).wait()      // wait() is synchronous!!!
```

Comment:
Couldn't the DNS resolution hang? Should we timeout after some time like 10 seconds?

Reply:
It seems that there's a built-in timeout of 10 seconds (either in the crate or at the OS level). I tried changing the nameserver to `99.99.99.99` in `/etc/resolv.conf` and it timed out:
```
Tue 2019-03-26 20:28:50.593652 [DEBUG] - Appending default port to "chococat.conwayste.rs"
Tue 2019-03-26 20:28:50.593708 [TRACE] - Resolving "chococat.conwayste.rs:2016"...
Tue 2019-03-26 20:29:00.605324 [ERROR] - failed to resolve: Custom { kind: Other, error: StringError("failed to lookup address information: Name or service not known") }
```

Reply:
But in most common situations, it fails almost instantly.